src/schema.rs

```diff
@@ -18,3 +18,40 @@ impl Schema {
         self.fields.insert(field.to_string(), typ);
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn normal_fields() {
+        let mut s = Schema::default();
+
+        s.add_field("str", Type::String);
+        s.add_field("ip", Type::IpAddr);
+        s.add_field("cidr", Type::IpCidr);
+        s.add_field("r", Type::Regex);
+        s.add_field("i", Type::Int);
+
+        assert_eq!(s.type_of("str"), Some(&Type::String));
```

Comment:
nit: it should be ok to implement `Copy` to `Type`, so that we don't need `&` here.